File: apps/query-burst/src/main/java/com/booster/queryburst/product/web/dto/response/ProductSummaryResponse.java
package com.booster.queryburst.product.web.dto.response;

import com.booster.queryburst.product.application.dto.ProductResult;
import com.booster.queryburst.product.domain.ProductStatus;

public record ProductSummaryResponse(
        Long id,
        String name,
        Long price,
        int stock,
        ProductStatus status,
        Long categoryId,
        String categoryName,
        Long sellerId,
        String sellerName
) {
    public static ProductSummaryResponse from(ProductResult result) {
        return new ProductSummaryResponse(
                result.id(),
                result.name(),
                result.price(),
                result.stock(),
                result.status(),
                result.categoryId(),
                result.categoryName(),
                result.sellerId(),
                result.sellerName()
        );
    }
}

File: apps/kotlin/shopping-service/src/main/kotlin/com/booster/kotlin/shoppingservice/catalog/web/dto/response/ProductSummaryResponse.kt
package com.booster.kotlin.shoppingservice.catalog.web.dto.response

import com.booster.kotlin.shoppingservice.catalog.domain.Product

data class ProductSummaryResponse(
    val id: Long,
    val name: String,
    val basePrice: Long,
    val status: String,
    val categoryId: Long,
    val thumbnailUrl: String?,
) {
    companion object {
        fun from(product: Product) = ProductSummaryResponse(
            id = product.id,
            name = product.name,
            basePrice = product.basePrice,
            status = product.status.name,
            categoryId = product.category.id,
            thumbnailUrl = product.images.find { it.isThumbnail }?.imageUrl,
        )
    }
}
